### PR TITLE
Migrating {project APIs, project DB collections} to auth server and adding grpc calls between graphql-server and auth server

### PR DESCRIPTION
**Pull request by SarthakJain26**
### Detailed info about the PR:
- Migrated project collection from litmus db to auth db, so as to prevent any DB inconsistencies and have smooth, efficient communication between users and project collection.
- Added APIs related to project collection in auth server and removed existing graphql APIs related to project and user management to have a better and cleaner code base.
- Added gRPC APIs in auth and graphql server so that both the servers can use RPC APIs to communicate internally.
- Added necessary frontend changes that were required due to additions and modifications in project related APIs

## Proposed changes

- Migrated project collection to auth DB
- Renamed `usercredentials` collection to `users` collection
- Added gRPC clients in graphql and auth server
- Modified usage query to work with new architecture
- Frontend integration with new APIs

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Review by S-ayanide — APPROVED** [review]

**Comment by S-ayanide** [comment]
Can you please fix the DeepScan issues!

File: litmus-portal/frontend/src/graphql/mutations.ts [hunk 1]
```diff
@@ -27,73 +27,73 @@ export const DELETE_WORKFLOW_TEMPLATE = gql`
...
-    acceptInvitation(member: $member)
-  }
-`;
-
-export const DECLINE_INVITE = gql`
-  mutation decline($member: MemberInput!) {
-    declineInvitation(member: $member)
-  }
-`;
+// export const CREATE_PROJECT = gql`
```
Comment (amityt) [inline litmus-portal/frontend/src/graphql/mutations.ts:47]: Can we remove the commented code if not required?

**Review by amityt — CHANGES_REQUESTED** [review]

File: litmus-portal/graphql-server/graph/resolver.go [hunk 2]
```diff
@@ -24,6 +24,11 @@ func NewConfig() generated.Config {
 			return nil, err
 		}
 		newCtx := context.WithValue(ctx, authorization.UserClaim, user)
+		// var x model.UsageQuery
+		// x.DateRange = &model.DateRange{
+		// 	StartDate: "1635942521",
+		// }
+		// _, err = usage.GetUsage(newCtx, x)
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/graph/resolver.go:31]: please remove

**Review by gdsoumya — CHANGES_REQUESTED** [review]

**Review by rajdas98 — APPROVED** [review]
LGTM, Checks are failing due to proto files, It will be ignored after this PR

**Review by gdsoumya — APPROVED** [review]

**Pull request merged**